### DISCUSSION TITLE
How can I recover a submodule that was deleted in Github?

### DISCUSSION THREAD

**Discussion by MarcoAlvarez97** [community/community#68812] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

I have deleted a submodule and also the entire folder, please, I need help, I don't know how to recover it

**Comment by vishalkrishna8** [comment]
Here are steps you can follow to restore it:

1.Check Git status to see deleted files.
2.Use `git restore` to unmark the submodule folder as deleted.
3.Re-add the submodule with `git submodule add`.
4.Initialize and update the submodule.
5.Commit changes with `git commit`.
6.If needed, push changes to the remote repository.
Let me know if this was helpful @MarcoAlvarez97
